packages/cubejs-query-orchestrator/src/orchestrator/PreAggregations.ts

```diff
@@ -1998,11 +1998,14 @@ export class PreAggregations {
   }
 
   public static targetTableName(versionEntry): string {
+    const dbCatalog = getEnv('dbCatalog');
+
+    const dbCatalogStringPart = dbCatalog ? `${dbCatalog}.` : '';
     if (versionEntry.naming_version === 2) {
-      return `${versionEntry.table_name}_${versionEntry.content_version}_${versionEntry.structure_version}_${encodeTimeStamp(versionEntry.last_updated_at)}`;
+      return `${dbCatalogStringPart}${versionEntry.table_name}_${versionEntry.content_version}_${versionEntry.structure_version}_${encodeTimeStamp(versionEntry.last_updated_at)}`;
     }
 
-    return `${versionEntry.table_name}_${versionEntry.content_version}_${versionEntry.structure_version}_${versionEntry.last_updated_at}`;
+    return `${dbCatalogStringPart}${versionEntry.table_name}_${versionEntry.content_version}_${versionEntry.structure_version}_${versionEntry.last_updated_at}`;
   }
```

Comment:
📓 required for external: false and temp table strategy, in any other case we will create table in hive_metastore by default